File: Processes/Python/exec_example.py
import os
import sys

pid = os.fork()
if pid < 0:
    print("fork failed", file=sys.stderr)
    sys.exit(1)
elif pid == 0:
    os.execvp("echo", ["echo", "Hello from exec"])
else:
    os.wait()
    print("Parent: exec finished")

File: Processes/C++/exec_example.cpp
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <iostream>
#include <cstdlib>

int main() {
    pid_t pid = fork();
    if (pid < 0) {
        std::cerr << "fork failed" << std::endl;
        return 1;
    } else if (pid == 0) {
        execlp("echo", "echo", "Hello from exec", nullptr);
        std::cerr << "execlp failed" << std::endl;
        return 1;
    } else {
        wait(nullptr);
        std::cout << "Parent: exec finished" << std::endl;
    }
    return 0;
}